sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/SerializerBuildHelper.scala

```diff
@@ -429,7 +430,10 @@ object SerializerBuildHelper {
         Literal(codecProvider(), ObjectType(classOf[Codec[_, _]])),
         "encode",
         externalDataTypeFor(encoder),
-        input :: Nil)
+        input :: Nil,
```

Comment:
It took me a while to realize that this fixes the nullability issue, and not the TransformingEncoder fix. Based on the nullability of the struct field we insert an AssertNotNull and a KnowNotNull; the correct nullability was not propagated by Invoke.

Reply:
I replaced the duplicative test with an explicit test for TransformingEncoder.nullable the catchily named "Encoder with OptionEncoder of transformation".  Encoding itself "works" but the output is incorrect with Some(null) vs None.  (NB in frameless this equates to 15 test failures)